### ISSUE TITLE
arm and arm64 binaries

### ISSUE THREAD

**Issue by patrickelectric** [svartalf/rust-battop#19]
It would be nice to have such binaries available

**Comment by the-jan-peter** [comment]
Yes, please.

**Comment by elzii** [comment]
Bump

**Comment by Martinligabue** [comment]
Steps I made to compile it myself. This is an ex Android phone with PostmarketOS:

<img width="1883" height="1001" alt="Image" src="https://github.com/user-attachments/assets/4ecb5b28-023f-48b3-8cf8-a08de089ddbb" />

In Arch I installed `aarch64-linux-gnu-gcc` and run `rustup target add aarch64-unknown-linux-gnu`

File Changes

    Cargo.toml: Update the [dependencies] section with specific versions of the battery and uom crates to resolve a version incompatibility error.

```
[dependencies]
battery = "0.7.8"
uom = "0.31.0"
```

.cargo/config.toml: Create this file to configure Cargo for aarch64 cross-compilation. This file tells Cargo to use the correct linker for your target device and to statically link all dependencies into a single binary.

```
    [target.aarch64-unknown-linux-gnu]
    linker = "aarch64-linux-gnu-gcc"
    rustflags = ["-C", "target-feature=+crt-static"]
```

Compilation Command

Once you have made the changes above, run this command to build the project. It will create the final executable in the target/aarch64-unknown-linux-gnu/release/ directory.

`cargo build --release --target aarch64-unknown-linux-gnu`

**Comment by elzii** [comment]
Yeah I got it to build on darwin aarch64 to using similar. Just annoying to have to temp-sabotage your rust toolchain when it could be done in a GH workflow instead. 🤷
